### PR TITLE
[MARTIFACT-35] Compare goal should fail build when comparison was uns…

### PR DESCRIPTION
**Pull request by michael-o**
…uccessful

### CODE REVIEW

File: src/main/java/org/apache/maven/plugins/artifact/buildinfo/CompareMojo.java [hunk 1]
```diff
@@ -241,6 +241,11 @@ private void compareWithReference( Map<Artifact, String> artifacts, File referen
         }
 
         copyAggregateToRoot( buildcompare );
+
+        if ( ko + missing > 0 )
+        {
+            throw new MojoExecutionException( "Build artifacts are not reproducible" );
```
Comment (hboutemy) [inline src/main/java/org/apache/maven/plugins/artifact/buildinfo/CompareMojo.java:247]: s/not reproducible/different from reference/

they may be reproducible if rebuilder fixes his environment, for example uses the right JDK, or changes anything that is required

Comment (michael-o) [reply src/main/java/org/apache/maven/plugins/artifact/buildinfo/CompareMojo.java]: fixed

**Comment by michael-o** [comment]
Same modified Doxia 2.0.0-M5 released:
```
PS C:\Entwicklung\Projekte\maven-doxia> mvn clean verify org.apache.maven.plugins:maven-artifact-plugin:3.3.1-SNAPSHOT:compare "-Dreference.repo=file:///C:/Temp/repo" "-Dspotless.check.skip"
...
[INFO] --- maven-artifact-plugin:3.3.1-SNAPSHOT:compare (default-cli) @ doxia-module-apt ---
[INFO] Reference buildinfo file not found: it will be generated from downloaded reference artifacts
[INFO] Reference build java.version: 1.8 (from MANIFEST.MF Build-Jdk-Spec)
[INFO] Reference build os.name: Windows (from pom.properties newline)
[INFO] Minimal buildinfo generated from downloaded artifacts: C:\Entwicklung\Projekte\maven-doxia\doxia-modules\doxia-module-apt\target\reference\doxia-module-apt-2.0.0-M5.buildinfo
[ERROR] size mismatch doxia-module-apt-2.0.0-M5.pom: investigate with diffoscope doxia-modules\doxia-module-apt\target\reference\doxia-module-apt-2.0.0-M5.pom doxia-modules\doxia-module-apt\pom.xml
[ERROR] size mismatch doxia-module-apt-2.0.0-M5.jar: investigate with diffoscope doxia-modules\doxia-module-apt\target\reference\doxia-module-apt-2.0.0-M5.jar doxia-modules\doxia-module-apt\target\doxia-module-apt-2.0.0-M5.jar
[ERROR] Reproducible Build output summary: 0 files ok, 2 different
[ERROR] see diff doxia-modules\doxia-module-apt\target\reference\doxia-module-apt-2.0.0-M5.buildinfo doxia-modules\doxia-module-apt\target\doxia-module-apt-2.0.0-M5.buildinfo
[ERROR] see also https://maven.apache.org/guides/mini/guide-reproducible-builds.html
[INFO] Reproducible Build output comparison saved to C:\Entwicklung\Projekte\maven-doxia\doxia-modules\doxia-module-apt\target\doxia-module-apt-2.0.0-M5.buildcompare
[INFO] Aggregate buildcompare copied to C:\Entwicklung\Projekte\maven-doxia\target\doxia-2.0.0-M5.buildcompare
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Doxia 2.0.0-M5:
[INFO]
[INFO] Doxia .............................................. SUCCESS [  7.035 s]
[INFO] Doxia :: Sink API .................................. SUCCESS [  2.935 s]
[INFO] Doxia :: Test Documents ............................ SUCCESS [  1.650 s]
[INFO] Doxia :: Core ...................................... SUCCESS [ 17.690 s]
[INFO] Doxia :: Modules ................................... SUCCESS [  0.505 s]
[INFO] Doxia :: APT Module ................................ FAILURE [  7.435 s]
[INFO] Doxia :: FML Module ................................ SKIPPED
[INFO] Doxia :: XDoc Module ............................... SKIPPED
[INFO] Doxia :: XHTML5 Module ............................. SKIPPED
[INFO] Doxia :: Markdown Module ........................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  39.460 s
[INFO] Finished at: 2023-02-03T15:47:27+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-artifact-plugin:3.3.1-SNAPSHOT:compare (default-cli) on project doxia-module-apt: Build artifacts are different from reference -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :doxia-module-apt
```

with `-fae`:
```
[INFO] Aggregate buildcompare copied to C:\Entwicklung\Projekte\maven-doxia\target\doxia-2.0.0-M5.buildcompare
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Doxia 2.0.0-M5:
[INFO]
[INFO] Doxia .............................................. SUCCESS [  7.795 s]
[INFO] Doxia :: Sink API .................................. SUCCESS [  2.857 s]
[INFO] Doxia :: Test Documents ............................ SUCCESS [  1.055 s]
[INFO] Doxia :: Core ...................................... SUCCESS [ 17.235 s]
[INFO] Doxia :: Modules ................................... SUCCESS [  0.615 s]
[INFO] Doxia :: APT Module ................................ FAILURE [  8.641 s]
[INFO] Doxia :: FML Module ................................ SUCCESS [  4.225 s]
[INFO] Doxia :: XDoc Module ............................... SUCCESS [  7.154 s]
[INFO] Doxia :: XHTML5 Module ............................. SUCCESS [  6.360 s]
[INFO] Doxia :: Markdown Module ........................... FAILURE [  7.380 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:05 min
[INFO] Finished at: 2023-02-03T15:49:44+01:00
[INFO] ------------------------------------------------------------------------
```

In my opinion, this is acceptable as well.

**Review by hboutemy — APPROVED** [review]
great, thanks

**Comment by jira-importer** [comment]
Resolve #119

**Pull request closed without merging**